tests/ArrayWrapper/ArrayWrapper.cpp

```diff
@@ -28,7 +28,7 @@ int main( int /*argc*/, char ** /*argv*/ )
 {
   vk::ArrayWrapper1D<char, 10> aw1( { 'f', 'o', 'o', 'b', 'a', 'h' } );
   std::string                  s1 = aw1;
-  assert( s1.length() == 10 );
+  assert( s1.length() == 6 );
```

Comment:
For completeness, I would add 2 more tests for boundary conditions:
1. ArrayWrapper1D of size 10 with 10 non-null-terminator characters initializing it. This should return size 10.
2. ArrayWrapper1D of size 10 with 9 non-null-terminator characters and an explicit null-terminator at index 9. This should return size 9.